spec/sections/copies.bs

```diff
@@ -170,19 +175,63 @@ dictionary GPUImageCopyTexture {
-        - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/x=] must be a multiple of |blockWidth|.
-        - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/y=] must be a multiple of |blockHeight|.
-        - The [=imageCopyTexture physical subresource size=] of |imageCopyTexture| is equal to |copySize| if either of
-            the following conditions is true:
-            - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/format}} is a depth-stencil format.
-            - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/sampleCount}} &gt; 1.
+
+        <div class=validusage>
+            - |imageCopyTexture|.{{GPUImageCopyTexture/texture}} must be a [=valid=] {{GPUTexture}}.
+            - |imageCopyTexture|.{{GPUImageCopyTexture/mipLevel}} must be &lt;
+                |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/mipLevelCount}}.
+            - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/x=] must be a multiple of |blockWidth|.
+            - |imageCopyTexture|.{{GPUImageCopyTexture/origin}}.[=GPUOrigin3D/y=] must be a multiple of |blockHeight|.
+            - The [=imageCopyTexture physical subresource size=] of |imageCopyTexture| is equal to |copySize| if either of
+                the following conditions is true:
+                - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/format}} is a depth-stencil format.
+                - |imageCopyTexture|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/sampleCount}} &gt; 1.
+        </div>
+</div>
+
+<div algorithm data-timeline=device>
+    <dfn abstract-op>validating texture buffer copy</dfn>(|imageCopyTexture|, |dataLayout|, |dataLength|, |copySize|, |aligned|)
+
+    **Arguments:**
+
+    - {{GPUImageCopyTexture}} |imageCopyTexture|
+    - {{GPUImageDataLayout}} |dataLayout|
+    - {{GPUSize64Out}} |dataLength|
+    - {{GPUExtent3D}} |copySize|
+    - {{boolean}} |aligned|
+
+    **Returns:** {{boolean}}
+
+    [=Device timeline=] steps:
+
+    1. Let |texture| be |imageCopyTexture|.{{GPUImageCopyTexture/texture}}
+    1. Let |aspectSpecificFormat| = |texture|.{{GPUTexture/format}}.
+    1. Let |offsetAlignment| = [=texel block copy footprint=] of |texture|.{{GPUTexture/format}}.
+
+    1. Return `true` if and only if all of the following conditions apply:
+
+        <div class=validusage>
+            1. [$validating GPUImageCopyTexture$](|imageCopyTexture|, |copySize|) returns `true`.
+            1. |texture|.{{GPUTexture/sampleCount}} is 1.
+            1. If |texture|.{{GPUTexture/format}} is a [=depth-or-stencil format=] format:
+                1. |imageCopyTexture|.{{GPUImageCopyTexture/aspect}} must refer to a single aspect of
+                    |texture|.{{GPUTexture/format}}.
+                1. That aspect must be a valid image copy source according to [[#depth-formats]].
```

Comment:
"source" makes this only work for texture->x validation and not x->texture validation

Reply:
Hmm... okay. I pushed a new version that solves this and pulls one more step into the common algorithm. I'm not a big fan of how many arguments this algorithm takes, but it seems like a reasonable tradeoff here? WDYT?

Reply:
Looks good! Pretty clean reuse of the usage.